tests/unit-tests/scene/test_surface_stack.cpp

```diff
@@ -1448,11 +1456,36 @@ TEST_F(SurfaceStack, observer_is_notified_on_lock)
     stack.lock();
 }
 
-TEST_F(SurfaceStack, observer_is_notified_on_unlock)
+TEST_F(SurfaceStack, observer_is_notified_only_on_initial_lock)
+{
+    auto observer = std::make_shared<MockSessionLockObserver>();
+    EXPECT_CALL(*observer, on_lock()).Times(1);
+    stack.register_interest(observer);
+    stack.lock();
+    stack.lock();
+    stack.lock();
+}
```

Comment:
The test body doesn't agree with the test name (you likely want to add a strategic `VerifyAndClearExpectations` after the first lock and set new expectations)